NuKeeper.Abstractions/Git/IGitDriver.cs

```diff
@@ -9,6 +9,8 @@ public interface IGitDriver
 
         void Clone(Uri pullEndpoint);
 
+        void Clone(Uri pullEndpoint, string branchName);
```

Comment:
I think a clone function should mimic `git clone` as closely as possible. Therefore adding a extra parameter `branchName` doesn't make sense to me.

Reply:
Git clone supports it:

something like:
git clone --single-branch --branch [branchname] [Git-Uri]